inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -260,15 +268,23 @@ private Optional<FlinkKafkaPartitioner<RowData>> getFlinkKafkaPartitioner(
             inLongFixedPartitionPartitioner.setSinkMultipleFormat(tableOptions.getOptional(SINK_MULTIPLE_FORMAT)
                     .orElse(null));
             return Optional.of(inLongFixedPartitionPartitioner);
-        }
-        if (tableOptions.getOptional(SINK_PARTITIONER).isPresent()
+        } else if (tableOptions.getOptional(SINK_PARTITIONER).isPresent()
                 && SINK_PARTITIONER_VALUE_RAW_HASH.equals(tableOptions.getOptional(SINK_PARTITIONER).get())) {
             RawDataHashPartitioner<RowData> rawHashPartitioner = new RawDataHashPartitioner<>(tableOptions
                     .getOptional(DATASOURCE_PARTITION_MAP).orElse(Collections.emptyMap()));
             rawHashPartitioner.setSinkMultipleFormat(tableOptions.getOptional(SINK_MULTIPLE_FORMAT).orElse(null));
             rawHashPartitioner.setPartitionPattern(tableOptions.getOptional(SINK_MULTIPLE_PARTITION_PATTERN)
                     .orElse(null));
             return Optional.of(rawHashPartitioner);
+        } else if (tableOptions.getOptional(SINK_PARTITIONER).isPresent()
+                && SINK_PARTITIONER_VALUE_PRIMARY_KEY.equals(tableOptions.getOptional(SINK_PARTITIONER).get())) {
+            SingleTablePrimaryKeyPartitioner<RowData> primaryKeyPartitioner = new SingleTablePrimaryKeyPartitioner<>();
+            // the pattern is different from the ${} pattern, it is a truncated string of schema fields.
+            primaryKeyPartitioner.setPartitionNumber(tableOptions.getOptional(SINK_FIXED_IDENTIFIER).orElse(null));
+            primaryKeyPartitioner.setPartitionKey(tableOptions.getOptional(SINK_MULTIPLE_PARTITION_PATTERN)
```

Comment:
you reuse SINK_MULTIPLE_PARTITION_PATTERN to parse partition, pattern define of SINK_MULTIPLE_PARTITION_PATTERN is different with your define. I suggest add description in `withDescription`.
```java
public static final ConfigOption<String> SINK_MULTIPLE_PARTITION_PATTERN =
            ConfigOptions.key("sink.multiple.partition-pattern")
                    .stringType()
                    .noDefaultValue()
                    .withDescription("option 'sink.multiple.partition-pattern' used when the partitioner is raw-hash.");
```
And update inlong document.

Reply:
where should I update the document? I can't find any description..
<img width="1766" alt="image" src="https://github.com/apache/inlong/assets/32808678/f5f4b243-9566-4163-a2b8-3bbe54dc9fef">

Reply:
description has been added